### ISSUE TITLE
Preserve Controller Templates set by user in Steam

### ISSUE THREAD

**Issue by sairuk** [doZennn/steam-rom-manager#383]
When refreshing the listing with SRM the existing Controller Options are reset instead of being reapplied to the entry, iirc these are stored in the vdf file so should be accessible.

Happy to know if there is an existing way to setup my parsers to carry these settings across, or if not, is this something that could be added to the project? Right now, its incredibly frustrating to have to reapply Controller Options after a refresh of the list.

This is currently what my intended state for titles added for each emu are

```
Steam Controller Legend:
d: Desktop Config
o: Overlay
i: Input Per Game

States
e: enabled // x: disabled

| game/platform               |  d  |  o  |  i  |
+-----------------------------+-----+-----+-----+
| MAME                        |  e  |  x  |  x  |
| Cloud Gaming                |  x  |  e  |  e  |
| Dolphin                     |  e  |  x  |  x  |
| DuckStation                 |  e  |  x  |  x  |
| PCSX2                       |  e  |  e  |  x  |
| Supermodel                  |  e  |  x  |  x  |
| Flycast                     |  e  |  e  |  e  |
| Kega Fusion                 |  e  |  x  |  x  |
```

if its of any consequence
- SRM Flatpak 2.3.40
- Steam Beta
- Manjaro Linux
- Big Picture Mode (Primary interface)

**Comment by cbartondock** [comment]
This is at least partially fixed now by the ability to set controller templates within SRM, although I do understand that that's not quite your issue. You'd like ones you set within steam to be preserved by SRM.

You're right it should be do-able.

**Comment by cbartondock** [comment]
Did this happen every time you refreshed the list or only when you changed the game name /path? Because I don't actually think SRM does reset controller configs if nothing about the game changes.

**Comment by sairuk** [comment]
Appreciate you looking into this, it has been a while but I can and add and rebuild a test user to confirm whatever behaviour will help out. Machine is deployed with ansible and when I raised this it was when old BPM was relevant, I have not rebuilt since new BPM was released as I couldn't be bothered breaking the setup and reconfiguring 50 games manually.

Nothing is changing here apart from rebuilding the list when adding additional games. I do however use .desktop files (generated with ansible) and the steamgrid configs deployed in the same way because i have multiple users (8) on the system. I then load the flatpak version of SRM and select generate. SRM does not appear to query the older titles so I figured it was not detecting them as new. Perhaps it is?

how does this tool detect new titles on list generation, is it using created/mod date? I will confirm ansible isn't force changing the generated file mod date forcing SRM to detect the entry as new.

Is this behaviour incidental because i am using .desktop files as my basis and they are queried and always determined as new?
